cms/admin/pageadmin.py

```diff
@@ -1420,17 +1384,50 @@ def get_tree(self, request):
         pages = self.get_queryset(request)
         node_id = request.GET.get('nodeId')
         open_nodes = list(map(int, request.GET.getlist('openNodes[]')))
+        node_page = None
+        root_pages = None
+
+        # Retrieve search parameters if exist
+        query = request.GET.get('q', None)
+        changelist_form = self.changelist_form(request.GET)
+
+        try:
+            changelist_form.full_clean()
```

Comment:
I'm not convinced that this logic should have been removed from the ChangeList view, it is directly associated with the Changelist view where as get_tree can be used in isolation i.e. it's not dependant on the Changelist view. With this change get_tree is now Changeview dependant when it shouldn't be.

Reply:
As `changelist_form` is a class attribute, I considered its existence intrinsic to `BasePageAdmin` which contains also `get_tree` method so from my point of view, we could always use `ChangeListForm` into `get_tree`.

I used it as a helper to get pages from query params pass to `get_tree` methods but I could also retrieve pages by executing manually queries without using a `ChangeListForm`. WDYT?